staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -82,26 +84,50 @@ func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool,
 func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		return f.checkClient(ctx, c)
 	default:
 		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, c client) error {
+	go func() {
+		// start with 10 ms, multiply by 2 each step, 16 steps ~= 10 minutes
+		backoff := wait.Backoff{
+			Duration: 10 * time.Millisecond,
+			Jitter:   0,
+			Factor:   2,
+			Steps:    16,
+		}
+		err := wait.ExponentialBackoff(backoff, func() (done bool, err error) {
```

Comment:
Also please let me know if i use the right function with right parameters, it seems ok to me.